forc-test/src/lib.rs

```diff
@@ -151,16 +343,61 @@ impl BuiltTests {
 
 /// First builds the package or workspace, ready for execution.
 pub fn build(opts: Opts) -> anyhow::Result<BuiltTests> {
-    let build_opts = opts.into_build_opts();
-    let built_tests = match pkg::build_with_options(build_opts)? {
-        pkg::Built::Package(pkg) => BuiltTests::Package(pkg),
-        pkg::Built::Workspace(workspace) => {
-            BuiltTests::Workspace(workspace.values().cloned().collect())
-        }
-    };
+    let build_opts = opts.clone().into_build_opts();
+    let built = pkg::build_with_options(build_opts)?;
+    let built_tests = BuiltTests::from_built(built, opts)?;
     Ok(built_tests)
 }
 
+/// Deploys the provided contract and returns an interpreter instance ready to be used in test
+/// executions with deployed contract.
+fn deploy_test_contract(built_pkg: BuiltPackage) -> anyhow::Result<TestSetup> {
+    // Obtain the contract id for deployment.
+    let mut storage_slots = built_pkg.storage_slots;
+    storage_slots.sort();
+    let bytecode = built_pkg.bytecode;
+    let contract = tx::Contract::from(bytecode.clone());
+    let root = contract.root();
+    let state_root = tx::Contract::initial_state_root(storage_slots.iter());
+    let salt = tx::Salt::zeroed();
+    let contract_id = contract.id(&salt, &root, &state_root);
+    // TODO: Remove this prompt once https://github.com/FuelLabs/sway/issues/3673 is addressed.
+    println!(
+        "      Deploying contract with id {:?} for testing",
+        contract_id
+    );
+
+    // Setup the interpreter for deployment.
+    let params = tx::ConsensusParameters::default();
+    let storage = vm::storage::MemoryStorage::default();
+    let mut interpreter = vm::interpreter::Interpreter::with_storage(storage, params);
+
+    // Create the deployment transaction.
+    let mut rng = rand::rngs::StdRng::seed_from_u64(0x7E57u64);
+    let metadata: TxMetadata = rng.gen();
+
+    let tx = tx::TransactionBuilder::create(bytecode.into(), salt, storage_slots)
+        .add_unsigned_coin_input(
+            metadata.secret_key,
+            metadata.utxo_id,
+            metadata.amount,
+            metadata.asset_id,
+            metadata.tx_pointer,
+            metadata.maturity,
+        )
+        .add_output(tx::Output::contract_created(contract_id, state_root))
+        .maturity(metadata.maturity)
+        .finalize_checked(metadata.block_height, &params);
+
+    // Deploy the contract.
+    interpreter.transact(tx)?;
```

Comment:
My initial approach was like that but interpreter instance does not let transacting 2 different types of transaction (`crate` and `script`) to be transact with the same instance. Because the instance is taking the type of the transaction it is runnnig. (https://github.com/FuelLabs/fuel-low-level/blob/bca7c506cdb1168266626b019ad95535a264292a/fuel-vm/src/transactor.rs#L149 might explain thing better. `fn transact()` creates the interpreter instance with the transactions type.) Not sure why that is implemented that way but to walk around that I had to create storages with contract deployed for each tests that requires them.